### PR TITLE
chore: bump dashpay/platform pin to PR #3968 tip (seedless rehydration + shielded viewing-key persistence)

### PR DESCRIPTION
**Pull request by Claudius-Maginificent**
**TL;DR:** Bump the pinned `dashpay/platform` dependency to the current tip of upstream PR #3968, landing seedless wallet rehydration and **native persistence for shielded (Orchard) viewing keys**, and adapting DET's wallet backend to further API changes as the branch has advanced. Closes review findings from a security/structural/adversarial due-diligence pass on each adaptation, across four rounds so far (initial bump review, a grumpy-review follow-up, a PR-comment triage follow-up, and a further re-bump + regression-fix + comment-triage round): one HIGH and one MEDIUM security issue, three LOW privacy leaks, one MEDIUM test-coverage gap, one MEDIUM cold-boot resilience regression (an intentional upstream behavior change, accepted and given an accurate error message rather than worked around), and several LOW/MEDIUM nits — all fixed and verified before merge. **Correction (2026-07-23):** a follow-up grumpy-review pass found this description previously (and incorrectly) claimed "no user-observable behavior change" — see "User-observable changes" below for what actually changed.

## User story
As a **DET user with a password-protected wallet that sends or receives shielded (Orchard) funds**, I want the wallet's underlying platform dependency updated with upstream's latest persistence and security improvements, so my shielded funds stay fully recoverable from my seed phrase and my data directories are properly protected — even though it means a stricter password floor and an extra passphrase prompt per shielded transaction.

### Base flow
A user creates a wallet (optionally password-protected), uses it for transparent and shielded sends/receives/withdrawals, and expects that data to survive app restarts and be fully recoverable from the seed phrase alone.

### Actual behavior (before this PR)
- Wallet passwords had no enforced minimum length.
- Once a protected wallet was unlocked, its shielded spend key stayed session-resident, so a passphrase prompt appeared only once per session no matter how many shielded operations followed.
- Removing a wallet could still leave its shielded viewing key behind in an automatic backup snapshot, even though the user was told the wallet's data was cleared.

### Expected behavior (after this PR)
- New wallet passwords must be at least 8 UTF-8 bytes.
- A protected wallet now prompts for its passphrase on every shielded transfer/unshield/withdraw, because the spend key is resolved just-in-time per operation and dropped immediately after — private key material is never left resident between operations.
- Removing a wallet fully deletes its shielded viewing key, including from automatic backups.
- Wallets already migrated to Tier-2 storage by a narrow, unreleased July-2026 weekly build with a shorter password cannot be opened at this pin — a documented compatibility note for that cohort only.

### Shielded module changes
This bump changes how DET handles shielded (Orchard) key material end-to-end:
- **Native FVK persistence**: upstream `platform-wallet-storage` now persists each wallet's shielded Full Viewing Key (FVK) natively in its own `shielded_viewing_keys` table, replacing DET's bespoke `DetPersister` metadata adapter (retired once upstream gained this support). This is what makes seedless shielded rehydration possible: DET can resync/rebind shielded state on a later launch without re-entering a passphrase.
- **Everything is still 100% derived from the wallet's single BIP-39 seed** — no independent or separately-stored key material. Orchard spend authority is derived from the seed via the standard ZIP-32 hierarchical path (`m / 32' / coin_type' / account'`); only the FVK (view-only, cannot authorize a spend) is what's cached to disk, computed once at bind time. The spend authority itself is re-derived from the seed transiently on every shielded operation and dropped immediately after — never persisted. Losing the persisted FVK costs a rescan, not funds; restoring from seed phrase alone recovers everything. (Verified against the pinned `orchard`/`rust-dashcore`/`platform-wallet` sources — see the PR discussion for the full derivation-path citation.)
- **JIT (just-in-time) seed resolution changes user-facing behavior**: `shielded_transfer`/`shielded_unshield`/`shielded_withdraw` now resolve the HD seed through the secret-seam chokepoint fresh for each single spend (`src/wallet_backend/shielded.rs`), rather than operating on an already-bound, session-resident spend key as before. Net effect: a password-protected (Tier-2) wallet now prompts for its passphrase on **every** shielded spend, not once per session — see "User-observable changes" below and `docs/MCP.md`/`docs/CLI.md` for the headless/MCP automation implication.
- **Residual, accepted risk**: FVKs are stored unencrypted at rest, consistent with upstream's own "viewing-grade, not secret" classification (a viewing key can't move funds) — not a blocker, a documented trade-off.
- **Privacy fix, this update**: removing a wallet now uses upstream's no-backup deletion entry point, so the shielded viewing key doesn't survive in an automatic pre-delete backup snapshot after the user is told the wallet was removed (see "PR-comment triage follow-up" below).

### User-observable changes
This is primarily an internal dependency bump, but it is **not** behavior-neutral. Three things a user can notice:
- **New wallet passwords must be at least 8 UTF-8 bytes** (measured in bytes, not Unicode characters — so a 4-character non-ASCII password like `öäüß`, 8 bytes, is accepted) — creation is rejected below that floor.
- **Wallets already migrated to Tier-2 storage by a July-2026 weekly build with a shorter password cannot be opened at this pin** — a documented compatibility blocker for that narrow, unreleased cohort (see CHANGELOG).
- **Protected (Tier-2) shielded wallets now prompt for a passphrase on every shielded spend** (transfer/unshield/withdraw), not just once per session — see `docs/MCP.md`/`docs/CLI.md` for the headless/MCP automation implication.

### What was done
- Bumped `dash-sdk`, `rs-sdk-trusted-context-provider`, `platform-wallet`, `platform-wallet-storage` git pins, cumulatively `d18020f5` → `f376d32b` → `ebbd15c4` → `e75f259f` → `4a1ba64c` → `288a6cae` (`dashpay/platform` PR #3968 tip, branch `feat/platform-wallet-storage-rehydration`). This is a re-bump within the same open PR: the branch has advanced substantially past each previous pin every time this update runs.
- **`4a1ba64c` → `288a6cae`** (2026-07-24): picked up mainline `v4.1-dev` merges (release/version bumps, a `kotlin-sdk` fix, the docker test-suite image, a `drive` proof wire-format versioning fix) — no direct changes to the four pinned crates' own code in this range. See "Re-bump + cold-boot regression fix + comment triage" below for what this pin move required on the DET side.
- **`ebbd15c4` → `e75f259f`** (previously landed, not yet reflected in this description until now): the hardened file and SQLite storage backends now validate permissions on every parent directory (app-data root, SPV dir, per-network dir normalized to owner-only on Unix); `PlatformWalletPersistence` gained `delete_wallet`, with `DetPersister` delegating to the inner `SqlitePersister` so wallet deletion keeps the typed backend behavior; the secret store enforces an eight-byte passphrase floor for new HD wallets while keeping unmigrated short-password legacy wallets usable; upstream now persists shielded viewing keys natively, with DET's legacy metadata adapter retained for databases written by prior pins.
- **`e75f259f` → `4a1ba64c`** (this update): two further API-drift adaptations forced by the upstream jump — `DataContractUpdate` state transitions now wait on `broadcast_and_wait_for_affected_state` instead of the generic `broadcast_and_wait`, matching upstream's reclassification of contract updates as non-execution-binding (a correctness fix, not a weaker proof: both paths run identical GroveDB + quorum-BLS verification and reject consensus errors first, only the tag-gate differs); and contact receiving-account registration (`register_contact_receiving_accounts`) now goes through `ManagedAccountOperations::add_managed_account_from_xpub` instead of a lower-level insert, because the nested `rust-dashcore` pin bump changed account-generation semantics so that adding a managed account now correctly invalidates stale SPV compact-filter coverage. Only extended *public* keys cross this seam — no seed or private key material is touched.
- `Cargo.lock` is regenerated scoped to the platform package set only at each bump — no crates.io registry version, checksum, or dependency-edge drift introduced.

### Due-diligence review
Independent security (Smythe) + structural (Adams) + adversarial (Marvin) review of each adaptation, before merge:
- **SEC-007 (HIGH, confirmed via reproduction):** `DetPersister::load()` originally propagated any single wallet's corrupt/undecodable FVK row as a hard error, failing cold-boot rehydration for **every** wallet in the store. Fixed to degrade a decode/schema/oversized-value failure to "absent" for the affected wallet only (logged), matching this module's existing `kv_get_logged` convention. Covered by a two-wallet, two-boot regression test. **Update (2026-07-23):** this fix and its test lived only in `DetPersister`, which a later commit in this same PR deleted once upstream gained native FVK support — see "Grumpy-review follow-up fixes" below; the underlying protection is real (now provided by upstream) but DET's own regression coverage for it had lapsed until this update restored it.
- **RUST-001 (MEDIUM):** the mixed-changeset reject guard inspected the nested `ShieldedChangeSet` by field access instead of exhaustive destructuring. Fixed to destructure all six fields.
- **SEC-008/SEC-010 (LOW, privacy-only):** removing a wallet never reaped its persisted FVK row, orphaning an Orchard viewing key on disk after wallet deletion. Fixed with a best-effort delete on wallet removal, covered by a regression test. **Update (2026-07-23):** that delete was fire-and-forget with no user-visible failure signal — fixed, see below.
- **QA-001 (MEDIUM, this update):** the initial regression test for the contact-account registration change only exercised the low-level upstream helper's account-generation/filter-invalidation behavior, never the actual `monitor_revision` gating in `register_contact_receiving_accounts` — the mechanism this change exists to protect (silent-missed-funds prevention via bloom-filter rebuild). Fixed: the gating logic was extracted into its own production seam (`register_contact_accounts_in_managed_wallet`), now covered by a regression proving first registration bumps the revision, a duplicate registration is a no-op, and a mixed existing/new batch bumps exactly once.
- **RUST-001 (LOW, this update):** the new `add_contact_receiving_account` helper lacked a doc comment, unlike its sibling free functions in the same file. Fixed — documents the `DashpayReceivingFunds`-only precondition and the boolean-return meaning.
- Two LOW code-quality questions from this update's review were resolved with concrete engineering answers rather than blind edits: the defensive non-`DashpayReceivingFunds` branch is kept (one-line comment: validated at this seam so future callers can't register the wrong account type), and DET's own duplicate-key pre-check is kept (upstream returns `InvalidParameter` on a duplicate insert rather than an idempotent no-op, so DET's check is load-bearing, not redundant).
- Residual accepted/deferred (LOW, non-blocking): FVKs are stored unencrypted at rest, consistent with upstream's own "viewing-grade, not secret" classification; this update's `update_data_contract.rs` broadcast-API switch has no dedicated automated test (verified correct by direct comparison against upstream's verifier source, but flagged rather than covered in this round); a live-testnet smoke test of upstream's tightened freshness-anchor enforcement remains a recommended follow-up (not covered by the committed lib-test ledger, and backend-E2E is excluded from CI).

### Grumpy-review follow-up fixes (2026-07-23)
A full 3-agent grumpy-review pass (security/project-consistency/QA) plus coordinator validation against live source found the description above overstated what this PR verifies and ships cleanly. 6 blocking + 3 non-blocking findings were fixed in a follow-up commit:
- **Corrected this description's false "no user-observable behavior change" claim** — see "User-observable changes" above — and updated `docs/user-stories.md` (WAL-001/002) to document the new password floor.
- **Restored DET-side regression coverage for cold-boot resilience to a corrupt shielded-FVK row**, exercising the real seedless cold-boot path (the original DET-owned fix/test were deleted when `DetPersister` was retired; the underlying protection now lives upstream, but DET had zero test pinning it down until this fix).
- **Fixed a real password-floor bug**: the new 8-character floor counted Unicode *characters* instead of UTF-8 *bytes*, wrongfully rejecting valid non-ASCII passwords upstream would accept (e.g. `"öäüß"` — 4 chars, 8 bytes). Now measured in bytes, matching upstream exactly, with a regression test.
- **Made wallet-deletion's shielded-FVK cascade failure visible**: it was previously fire-and-forget (a background-task failure only logged, never surfaced), silently leaving an unencrypted viewing key on disk after the user was told the wallet was removed. Now raises a persistent warning banner with technical details on failure, covered by a fault-injection regression test.
- **Documented the per-operation shielded-spend passphrase prompt's interaction with headless/MCP automation** in `docs/MCP.md`/`docs/CLI.md` — standalone/headless transports cannot prompt and will surface `SecretPromptUnavailable`; use an unprotected wallet for that automation for now.
- Reworded the new password-floor error messages (`WalletPasswordTooShort`) to match the jargon-free, actionable sibling convention already used elsewhere in this codebase.
- Narrowed the lazy Tier-2 re-wrap's error handling to only swallow the documented too-short-password case, propagating any other storage error (I/O faults, corruption) instead of silently absorbing it.
- Fixed a misleading error mapping (`hd_seed_for_shielded_spend`'s structural-misconfiguration guard returned `WalletLocked`, telling the user to "unlock" a wallet that wasn't locked) — now returns `WalletStateInconsistent`.
- Hardened `ensure_data_dir_exists` to create new data directories atomically at owner-only permissions (no create-then-chmod TOCTOU window) and to reject following a planted symlink on the existing-directory repair path.

### PR-comment triage follow-up (2026-07-23, round 2)
A pass over CodeRabbit/thepastaclaw's inline review threads on this PR found three more issues, all fixed:
- **BLOCKING privacy leak**: wallet deletion used the backup-taking `delete_wallet` entry point, so despite the SEC-008/SEC-010 fix above, an automatic pre-delete backup snapshot still preserved a copy of the removed wallet's unencrypted shielded viewing key on disk. Switched to `delete_wallet_skip_backup` (the pinned no-backup entry point for exactly this case); regression test extended to assert no auto-backup directory is created.
- **Missing repaint**: the async wallet-removal failure warning banner only wrote state into the egui context — without `ctx.request_repaint()`, an idle UI wouldn't paint it until an unrelated repaint happened to occur. Fixed; new unit test asserts a repaint is requested.
- **`bump_monitor_revision` over-firing**: `register_contact_accounts_in_managed_wallet` was bumping the monitor revision of *every* `DashpayReceivingFunds` account whenever any single one was new, contradicting its own doc comment and causing needless bloom-filter/rescan churn on already-covered contacts every time one new contact was added. Fixed to bump only the specific newly-inserted account; the existing regression test's assertions were corrected accordingly (it had been asserting the old, over-firing behavior) and strengthened to check per-account revisions individually.
- Also replied to (but did not apply) a hardcoded-test-password nitpick: the three flagged sites all predate this PR or belong to an unrelated PR merged into this branch, and use test-only placeholder strings, not real secrets.
- CHANGELOG.md wording tightened: password floor now says "UTF-8 bytes" throughout (was inconsistently "characters" in one place), the per-op shielded passphrase prompt got its own user-facing bullet, and the Tier-2 lockout compatibility note was un-buried from a 40-line paragraph into its own line.

Deliberately left out of both follow-up rounds (documented, not silently dropped): the unmerged-upstream-pin risk (inherent to this PR chain, resolves once #3968 merges), unencrypted-at-rest FVKs (consistent with upstream's own "viewing-grade, not secret" classification), the Tier-2 lockout's error-message clarity, a dedicated test for the `broadcast_and_wait_for_affected_state` switch, an upfront password-floor UI hint, one flaky test (`concurrent_spv_start_failure_is_returned_to_every_caller`, unrelated to this PR's content, needs separate concurrency diagnosis), and the pre-existing hardcoded test passwords noted above.

### Re-bump to 288a6cae + cold-boot regression fix + comment triage (2026-07-24)

Re-bumped the pin from `4a1ba64c` to `288a6cae` (upstream mainline `v4.1-dev` merges only — see "What was done" above). This surfaced one genuine regression and prompted a full pass over the PR's remaining open review comments.

- **Cold-boot resilience regression, accepted upstream behavior, fixed on the DET side**: the new pin ships a stricter upstream persister — `platform-wallet-storage`'s `core_state::load_state` now validates the `core_transactions.txid` column's fixed 32-byte width on every row and fails the entire cold-boot wallet load hard on a malformed row, where the prior pin never read that column at load time at all (a corrupt row was silently untouched). Confirmed by bisection this is a genuine, intentional upstream tightening, not a DET-side bug. Decision: accept upstream's fail-hard design rather than work around it. Rewrote the now-stale regression test (`cold_boot_keeps_wallet_visible_when_persisted_transaction_txid_is_corrupt` → `cold_boot_surfaces_typed_error_when_persisted_transaction_txid_is_corrupt`) to assert the new behavior, and added a dedicated `TaskError::WalletLocalDataLoadFailed` with an accurate, actionable message ("Saved wallet data appears damaged and cannot be loaded. Restore the wallet from its recovery phrase to keep using it.") — the prior generic "please retry" wrapper was actively misleading here, since this corruption is permanent and retrying fails identically every time. Scoped narrowly to the `Fatal` `PersisterLoad` kind only; every other persister-error classification is unchanged.
- **Six further review-comment fixes**, all from this PR's own open threads:
  - Wallet removal's local secret wipe (`forget_wallet_local_state`: seed vault, session cache, shielded rows) now surfaces a failure via the same warning banner the async upstream-removal step already used, instead of only logging and reporting a clean removal.
  - The data-contract update path now asserts the platform-returned contract's id matches the contract that was updated before caching it, rejecting a mismatch via a new `TaskError::UpdatedContractIdMismatch` instead of trusting the response blindly.
  - `docs/MCP.md`, `docs/CLI.md`, and `CHANGELOG.md` generalized the protected-wallet caveat from naming only shielded transfer/unshield/withdraw to all six operations that actually resolve the seed just-in-time (`src/wallet_backend/shielded.rs`'s `with_secret_session` call sites): initialization, shield-from-Core, shield-from-Platform, transfer, unshield, withdraw.
  - `WalletPasswordTooShort`/`PasswordTooShort` messages and their pinned display tests reworded from "characters" to "UTF-8 bytes after trimming", matching the byte-based enforcement and the CHANGELOG/docs wording already in place.
- **Two threads left open, deliberately**: a CodeRabbit hardcoded-test-password flag (`"short"`/`"legacy-pass"` in test fixtures) is disputed as a false positive — non-sensitive placeholders, not real secrets, replied on the thread; and the pre-existing `broadcast_and_wait_for_affected_state` test-coverage gap (already disclosed above) remains deferred — proper coverage needs SDK-mocking test infrastructure this file doesn't have yet, disproportionate scope for this round.

### Testing (2026-07-24 update)
- `cargo fmt --all` and `cargo fmt --all -- --check` clean at `288a6cae`.
- Targeted scopes on the fully merged tree: `context::wallet_lifecycle` (68 passed, including both the renamed cold-boot test and the new removal-warning test by name), `backend_task::error` (112 passed), `backend_task::update_data_contract` (2 passed, the new contract-id-guard tests), all green.
- `cargo clippy --all-features --all-targets -- -D warnings` clean.
- `Cargo.lock` diff for the `4a1ba64c` → `288a6cae` move confirmed scoped to the same 28 platform git packages only (`4.0.0` → `4.1.0-rc.1`); no crates.io registry drift.
- CI: Clippy, Clippy Report, CodeRabbit, Test Suite, and review all pass on the pushed commit.

### Testing
- `cargo fmt --all`, targeted `wallet_backend` (330 passed) and `backend_task` (411 passed) lib scopes, and `cargo clippy --lib --all-features -- -D warnings` all clean against the `4a1ba64c` pin.
- The new/updated regression (`contact_registration_bumps_monitor_revision_only_for_new_keys`) confirmed executing (not a zero-match false pass) with an explicit `... ok` in the verification log.
- `Cargo.lock` diff for this update confirmed scoped to the platform package set only: 28 platform git packages moved to `4a1ba64c`, 11 nested `rust-dashcore` packages moved `0091c4a` → `18c68d4`; no crates.io registry drift.
- Grumpy-review follow-up commit: `cargo fmt --all -- --check`, scoped `cargo clippy --lib --all-features -- -D warnings`, and targeted test scopes (`context::wallet_lifecycle` 65 passed, `model::wallet` 99 passed, `wallet_backend` 333 passed, `app_dir` 4 passed, `backend_task::error` 112 passed) all clean, independently re-verified by the coordinator with new/renamed test names confirmed present and passing in the logs (not just an aggregate count).
- PR-comment triage follow-up commit: same fmt/clippy gates clean; `wallet_backend` (337 passed) and `context::wallet_lifecycle` (67 passed) re-run, with `contact_registration_bumps_monitor_revision_only_for_new_keys` and the two new tests (`wallet_data_removal_warning_requests_repaint`, the extended `remove_wallet_reaps_persisted_shielded_viewing_keys`) confirmed present and passing by name in the logs, independently re-verified by the coordinator.

### Breaking changes
None for DET consumers — the four crates' public API is additive-only at this pin.

### Prior work
- Upstream: [dashpay/platform#3968](https://github.com/dashpay/platform/pull/3968) — the PR this pin tracks; still open as of this update.
- Upstream debt to watch: [dashpay/platform#3992](https://github.com/dashpay/platform/issues/3992) — a pre-existing manifest/`wallet_id` authentication gap, unrelated to this bump.

### Attribution
<sub>🤖 Co-authored by [Claudius the Magnificent](https://github.com/lklimek/claudius) AI Agent</sub>

## Summary by CodeRabbit

- **New Features**
  - Wallet data, shielded viewing keys, key pools, and DashPay accounts now persist and restore across restarts.
  - Shielded spending derives signing authority only when needed; protected wallets prompt for their passphrase for each operation.
  - Wallet passwords must contain at least 8 trimmed UTF-8 bytes, while legacy shorter-password wallets remain supported.

- **Bug Fixes**
  - Wallet deletion now removes persisted shielded data and displays a warning if cleanup fails.
  - Improved transaction confirmation, address synchronization, and recovery from transient startup issues.

- **Documentation**
  - Clarified protected-wallet behavior and limitations for standalone CLI and MCP usage.

### CODE REVIEW

**Comment by thepastaclaw** [comment]
🕓 Ready for review — 12 ahead in queue (commit 2b2da630)
_Queue position: 13/13 · 2 reviews active_
_ETA: start ~19:36 UTC · complete ~20:00 UTC (median 23m across 30 recent reviews; 2 slots)_
_Queued 6m ago · Last checked: 2026-07-24 17:10 UTC_

File: src/wallet_backend/mod.rs [hunk 1]
```diff
@@ -1372,11 +1375,20 @@ impl WalletBackend {
...
-            }),
+            Ok(_) | Err(PlatformWalletError::WalletNotFound(_)) => {}
+            Err(source) => {
+                return Err(TaskError::WalletBackend {
+                    source: Arc::new(source),
+                });
+            }
+        }
+
+        match self.inner.wallet_persister.delete_wallet(*wallet_id) {
```
Comment (thepastaclaw) [inline src/wallet_backend/mod.rs]: **🔴 Blocking: Wallet deletion preserves the viewing key in an automatic backup**

`WalletBackend::new` opens this persister with `SqlitePersisterConfig::new`, whose default enables automatic backups, and the pinned `delete_wallet` implementation takes a full pre-deletion database backup under the network database directory's `backups/auto` subdirectory before cascading wallet-owned rows. That snapshot includes the newly native, unencrypted `shielded_viewing_keys` row, so the Remove Wallet flow deletes the key from the live database while retaining a copy on disk despite telling the user that this version's local wallet data will be cleared. Use the pinned no-backup deletion entry point for this explicit user-requested removal, and extend the regression test beyond querying the live database so it detects creation of a pre-delete backup.

```suggestion
match self
            .inner
            .wallet_persister
            .delete_wallet_skip_backup(*wallet_id)
        {
```

<sub>source: ['codex']</sub>

File: src/model/wallet/mod.rs [hunk 2]
```diff
@@ -432,6 +436,11 @@ impl Wallet {
         // Encrypt seed or store plaintext
         let (encrypted_seed, salt, nonce, uses_password) = match password {
             Some(pw) if !pw.is_empty() => {
+                if pw.expose_secret().trim().chars().count() < MIN_PASSPHRASE_LEN {
```
Comment (thepastaclaw) [inline src/model/wallet/mod.rs]: **🟡 Suggestion: Match the storage policy's byte-length password check**

`MIN_PASSPHRASE_LEN` is the persistent store's post-trim byte-length floor, and upstream measures it with `SecretString::trimmed().len()`. This guard instead counts Unicode scalar values, so a password such as four two-byte non-ASCII characters satisfies the eight-byte storage requirement but is rejected by DET before storage is reached. Since this guard was introduced specifically to mirror the upstream floor, use Rust string byte length and add a multibyte boundary case to the existing short-password test.

```suggestion
if pw.expose_secret().trim().len() < MIN_PASSPHRASE_LEN {
```

<sub>source: ['codex']</sub>

**Review by thepastaclaw — CHANGES_REQUESTED** [review]
## Preliminary review — Codex only

The dependency adapters generally match the pinned upstream APIs, including the corrected contract-update wait path and contact-account revision gating. One blocking privacy defect remains because user-requested wallet deletion creates a pre-deletion backup containing the unencrypted shielded viewing key; the local password guard also rejects some passwords that satisfy the upstream byte-length policy.

Validated blockers were found in the Codex precheck. Sonnet is deferred until a fresh Codex revalidation clears the blocker gate.

### Review provenance
- Codex reviewers: `gpt-5.6-sol` — general (completed), `gpt-5.6-sol` — rust-quality (completed)
- Verifier: `gpt-5.6-sol` — verifier
- Sonnet: **not run (deferred by blocker gate)**

🔴 1 blocking | 🟡 1 suggestion(s)

<details>
<summary>🤖 Prompt for all review comments with AI agents</summary>

```
These findings are from an automated code review. Verify each finding against the current code and only fix it if needed.

In `src/wallet_backend/mod.rs`:
- [BLOCKING] src/wallet_backend/mod.rs:1389: Wallet deletion preserves the viewing key in an automatic backup
  `WalletBackend::new` opens this persister with `SqlitePersisterConfig::new`, whose default enables automatic backups, and the pinned `delete_wallet` implementation takes a full pre-deletion database backup under the network database directory's `backups/auto` subdirectory before cascading wallet-owned rows. That snapshot includes the newly native, unencrypted `shielded_viewing_keys` row, so the Remove Wallet flow deletes the key from the live database while retaining a copy on disk despite telling the user that this version's local wallet data will be cleared. Use the pinned no-backup deletion entry point for this explicit user-requested removal, and extend the regression test beyond querying the live database so it detects creation of a pre-delete backup.

In `src/model/wallet/mod.rs`:
- [SUGGESTION] src/model/wallet/mod.rs:439: Match the storage policy's byte-length password check
  `MIN_PASSPHRASE_LEN` is the persistent store's post-trim byte-length floor, and upstream measures it with `SecretString::trimmed().len()`. This guard instead counts Unicode scalar values, so a password such as four two-byte non-ASCII characters satisfies the eight-byte storage requirement but is rejected by DET before storage is reached. Since this guard was introduced specifically to mirror the upstream floor, use Rust string byte length and add a multibyte boundary case to the existing short-password test.
```
</details>

File: src/wallet_backend/secret_access.rs [hunk 3]
```diff
@@ -2637,6 +2644,45 @@ mod tests {
         );
     }
 
+    /// A legacy wallet may use a password shorter than the upstream Tier-2
+    /// floor. Successful legacy decryption must still release the seed for the
+    /// requested operation; the unsupported re-wrap is deferred and the
+    /// legacy envelope remains the source of truth.
+    #[tokio::test]
+    async fn short_legacy_seed_password_remains_usable_without_tier2_migration() {
+        const SHORT_LEGACY_PASSWORD: &str = "short";
```
[the remark this answers is not in the document]
Comment (Claudius-Maginificent) [reply src/wallet_backend/secret_access.rs:2678]: Checked all three sites — none are introduced by this PR and none are real secrets:

- `src/wallet_backend/secret_access.rs:2653` (`SENTINEL_PASSPHRASE`): pre-existing test constant, authored 2026-07-17, unrelated to this PR's diff.
- `src/boot.rs:297` / `src/boot.rs:332` (`"legacy-pass"`): arrived via the `v1.0-dev` merge into this branch (commit `69b7546eda`, 2026-07-22), authored on unrelated PR #927 — not part of this PR's own changes.

All three are test-only placeholder strings used to exercise password-length/legacy-migration logic in unit tests, not plaintext recovery phrases, private keys, or API tokens — the coding guideline's actual target. Given they're out of scope for this PR's diff, leaving them as-is rather than touching unrelated code here. Happy to file a follow-up if there's appetite for switching these to runtime-generated passphrases repo-wide.

**Comment by Claudius-Maginificent** [comment]
Also fixed the `bump_monitor_revision` nitpick from CodeRabbit's review summary (not a resolvable inline thread, noting it here for the record): `register_contact_accounts_in_managed_wallet` was bumping every `DashpayReceivingFunds` account's revision whenever any single one was new. It now bumps only the specific newly-inserted account, matching the function's own doc comment. The existing `contact_registration_bumps_monitor_revision_only_for_new_keys` test's final assertion was quietly wrong under the old behavior (`[1, 2]`, an artifact of over-bumping) and has been corrected to `[1, 1]` with per-account checks added. See commit `d99073aca`.

File: src/backend_task/update_data_contract.rs [hunk 4]
```diff
@@ -74,7 +74,10 @@ impl AppContext {
             .await
             .map_err(|e| TaskError::from(dash_sdk::Error::Protocol(e)))?;
 
-        match state_transition.broadcast_and_wait(sdk, None).await {
+        match state_transition
+            .broadcast_and_wait_for_affected_state(sdk, None)
```
[the remark this answers is not in the document]
Comment (Claudius-Maginificent) [reply src/backend_task/update_data_contract.rs]: Agreed this needs test coverage — deliberately deferred rather than dropped. Adding a proper mocked-SDK/e2e test for the `broadcast_and_wait_for_affected_state` swap needs test infrastructure `update_data_contract.rs` doesn't have yet, and that's disproportionate scope for a dependency pin-bump PR. Tracked as follow-up work; the PR description's due-diligence section already discloses this gap explicitly.

<sub>🤖 Co-authored by [Claudius the Magnificent](https://github.com/lklimek/claudius) AI Agent</sub>

File: src/wallet_backend/secret_access.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (Claudius-Maginificent) [reply src/wallet_backend/secret_access.rs:2678]: Assessed and disputed as a false positive: `SHORT_LEGACY_PASSWORD = "short"` and the `"legacy-pass"` literals are test-only fixture placeholders (used exclusively inside `#[cfg(test)]`/`#[tokio::test]` functions), not real secrets protecting any real wallet or funds — there's nothing to rotate. Leaving as-is; happy to switch to runtime-generated values if preferred, since that'd be a no-risk change, but not doing so proactively.

<sub>🤖 Co-authored by [Claudius the Magnificent](https://github.com/lklimek/claudius) AI Agent</sub>

**Comment by Claudius-Maginificent** [comment]
This pin regressed fetch_current: descending-epoch query now fails proof verification every sync, silently disabling shielded-tx gating + fee-multiplier refresh. Fix upstream: dashpay/platform#4231.

<sub>🤖 Co-authored by [Claudius the Magnificent](https://github.com/lklimek/claudius) AI Agent</sub>
